systemtest/src/main/java/io/strimzi/systemtest/Resources.java

```diff
@@ -741,6 +827,241 @@ DoneableKubernetesClusterRoleBinding kubernetesClusterRoleBinding(KubernetesClus
+                        .addToLabels("app", KAFKA_CLIENTS)
+                    .endMetadata()
+                    .withSpec(createClientSpec(tlsListener, kafkaUsers))
+                .endTemplate()
+            .endSpec()
+            .build();
+
+        return createNewDeployment(kafkaClient);
+    }
+
+    private static Service getSystemtestsServiceResource(String appName, int port) {
+        return new ServiceBuilder()
+            .withNewMetadata()
+                .withName(appName)
+                .addToLabels("run", appName)
+            .endMetadata()
+            .withNewSpec()
+                .withSelector(Collections.singletonMap("app", appName))
+                .addNewPort()
+                    .withName("http")
+                    .withPort(port)
+                    .withProtocol("TCP")
+                .endPort()
+            .endSpec()
+            .build();
+    }
+
+    DoneableService createServiceResource(String appName, int port, String clientNamespace) {
+        Service service = getSystemtestsServiceResource(appName, port);
+        LOGGER.info("Creating service {} in namespace {}", service.getMetadata().getName(), clientNamespace);
+        client.services().inNamespace(clientNamespace).create(service);
+        deleteLater(service);
+        return new DoneableService(service);
+    }
+
+    private static Ingress getSystemtestIngressResource(String appName, int port, String url) throws MalformedURLException {
+        IngressBackend backend = new IngressBackend();
+        backend.setServiceName(appName);
+        backend.setServicePort(new IntOrString(port));
+        HTTPIngressPath path = new HTTPIngressPath();
+        path.setPath("/");
+        path.setBackend(backend);
+
+        return new IngressBuilder()
+                .withNewMetadata()
+                .withName(appName)
+                .addToLabels("route", appName)
+                .endMetadata()
+                .withNewSpec()
+                .withRules(new IngressRuleBuilder()
+                        .withHost(appName + "." +  (ENVIRONMENT.getKubernetesDomain().equals(".nip.io") ?  new URL(url).getHost() + ".nip.io" : ENVIRONMENT.getKubernetesDomain()))
+                        .withNewHttp()
+                        .withPaths(path)
+                        .endHttp()
+                        .build())
+                .endSpec()
+                .build();
+    }
+
+    DoneableIngress createIngress(String appName, int port, String url, String clientNamespace) throws Exception {
+        Ingress ingress = getSystemtestIngressResource(appName, port, url);
+        LOGGER.info("Creating ingress {} in namespace {}", ingress.getMetadata().getName(), clientNamespace);
+        client.extensions().ingresses().inNamespace(clientNamespace).create(ingress);
+        deleteLater(ingress);
+        return new DoneableIngress(ingress);
+    }
+
+    private PodSpec createClientSpec(boolean tlsListener, KafkaUser... kafkaUsers) {
```

Comment:
Can we move this logic to a different location? `Resources` is a really big class =(

Reply:
You mean only this method? Guess it's possible to do that, but I don't see not enough benefit from it. Maybe it will be better to move out all methods from `Resources` which could be static and it doesn't matter from where they are called. You can create issue for that and we can manage it in some next PR.